File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {

        vector<vector<int>> mat;

        vector<int> ans;
        for (int i = 0; i < numRows; i++) {
            ans.push_back(1);
            mat.push_back(ans);
        }
        for (int i = 2; i < mat.size(); i++) {
            for (int j = 0; j < mat[i].size(); j++) {
                if (j == 0 || j == i)
                    continue;

                else {
                    mat[i][j] = mat[i - 1][j] + mat[i - 1][j - 1];
                }
            }
        }

        return mat ;
    }
};

File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {

        List<List<Integer>> list = new ArrayList<>();

        for (int i = 0; i < numRows ; i++) {
            ArrayList<Integer> temp = new ArrayList<>();
            int cur = 1;

            for (int j = 0; j <= i; j++) {

                temp.add(cur);
                cur = cur * (i - j) / (j + 1);

            }

            list.add(temp);

        }

        return list;

    }
}